addons/binding/org.openhab.binding.wink/src/main/java/org/openhab/binding/wink/client/DelegatedAuthenticationService.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.wink.client;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Uses a heroku app to manage access tokens, client ids and secrets for wink service. Removes the need for users
+ * to apply and use their own wink api app.
+ *
+ * To use, create the following wink.cfg
+ *
+ * auth_service = delegated
+ * auth_service_token = token_from_auth_service
+ *
+ * @author Shawn Crosby
+ *
+ */
+public class DelegatedAuthenticationService implements IWinkAuthenticationService {
+
+    private final Logger logger = LoggerFactory.getLogger(DelegatedAuthenticationService.class);
+
+    private String auth_token;
+    private String token;
+
+    public DelegatedAuthenticationService(String auth_token) {
+        this.auth_token = auth_token;
+        Client client = ClientBuilder.newClient();
+        WebTarget target = client.target("https://openhab-authservice.herokuapp.com");
```

Comment:
this is just a python application that I deployed to a heroku developer account that can be hosted for free.  I currently own it and I added a comment to the javadoc.